### PR TITLE
Major update: data consumer, data publisher, wis2node registration

### PR DESCRIPTION
**Pull request by 6a6d74**
plus updates of WIS2-SWIM interop section following ICAO review.

### CODE REVIEW

File: guide/sections/part1/data-publisher.adoc [hunk 1]
```diff
@@ -1,23 +1,204 @@
...
+
+Copies of all discovery metadata records from WIS2 are held at the Global Discovery Catalogues, where data consumers can search and browse to find data that is of interest to them. 
+
+Depending on local arrangements, your GISC may be able to help you transfer your discovery metadata record(s) to the Global Discovery Catalogues. If this is not the case, you will need to publish the discovery metadata record(s) yourselffootnote:[In future, WIS2 may provide metadata publication services (e.g., through a WIS2 metadata management portal) to assist with this task. However, such a service is not available at this time.] using one of two ways:
+
+* The simplest method is to encode the discovery metadata record as a file and publish it to an HTTP server where it can be accessed with a URL. 
+* Alternatively, you may operate a local metadata catalogue through which discovery metadata records can be shared using an API (e.g., OGC API - Recordsfootnote:[OGC API - Records - Part 1: Core https://docs.ogc.org/DRAFTS/20-004.html]). Each discovery metadata record can be accessed with a unique URL via the API (e.g., as an item which is part of thediscovery metadata catalogue).
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:34]: is part of the discovery metadata catalogue

File: guide/sections/part1/data-publisher.adoc [hunk 2]
```diff
@@ -1,23 +1,204 @@
...
+
+In both cases, a notification message needs to be published on a message broker that tells WIS2 there is new discovery metadata to upload and that it is accessed at the specified URLfootnote:[Both data and metadata publication use the same notification message mechanism to advertise the availability of a new resource.]. The notification messages shall conform to the WIS2 Notification Message (WNM) specification (TODO CROSSREF). Furthermore, the notification message must be published on topic that conforms to the WIS2 Topic Hierarchy (WTH) specification (TODO CROSSREF). For example, metadata published by Deutscher Wetterdienst would use the follow topic:
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:36]: metadata published by Deutscher Wetterdienst could use the following topic:

``origin/a/wis2/de-dwd/metadata/core``

File: guide/sections/part1/data-publisher.adoc [hunk 3]
```diff
@@ -1,23 +1,204 @@
...
+
+===== Providing data as files
+
+The simplest way to publish data through WIS2 is to persist your data as files and publish those files on a Web sserver. All these files need to be organised somehow – perhaps a flat structure or grouped into collections that resemble folders or directory structures.
+
+To make your data usable, your users need to be able to find the specific file (or files) they need.
+
+Naming conventions for files and/or directories are useful - but only when you understand the scheme. Assuming that your users understand your naming convention will be a barrier to widespread re-use; many users will simply treat the filename as an opaque string. Where communities commonly use file-naming conventions (e.g., with embedded metadata), these should only be used when adequate documentation is provided to users.
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:64]: only when the scheme is understood

File: guide/sections/part1/data-publisher.adoc [hunk 4]
```diff
@@ -1,23 +1,204 @@
...
+
+WIS2 does not require the use of specific naming conventions.
+
+Another mechanism to consider is complementing the collections (i.e., directories or folders in which files are grouped) with information that describes their content. Then users,both humans and software agents, can browse the structure and find what they need. Examples of this approach include:
+
+* Web Accessible Folders (WAF) and "readme" files: a Web-based folder structure listing the data object files by name, where each folder contains a formated "readme" file describing the folder contents.
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:70]: "README" files
"README" file

File: guide/sections/part1/data-publisher.adoc [hunk 5]
```diff
@@ -1,23 +1,204 @@
...
+. Interpret a user's request; 
+. Extract the data from wherever it is stored;
+. iii.	Package that data up and send it back to the user.
+
+Importantly, when considering use of interactive APIs to serve your data you need to plan for costs: every request to an interactive API requires computational resources to process.
+
+Based on the experience of data publishers who have been using Web APIs to serve their communities, this Guide makes the following recommendations about interactive APIs:
+
+* First, interactive APIs should be self-describing. A Data Consumer should not need to know, apriori, how to make requests from a API. They should be able to discover this information from the API endpoint itself – even if this is just a link to a documentation page they need to read.
+* Second, APIs should comply with OpenAPIfootnote:[OpenAPI Specification https://swagger.io/specification/] version 3 or later. OpenAPI provides a standardised mechanism to describe the API. Tooling (free and, commercial, etc.) is widely available that can read this metadata and automatically generate client applications to query the API.
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:90]: OpenAPI ref: https://spec.openapis.org/oas/v3.1.0

File: guide/sections/part1/data-publisher.adoc [hunk 6]
```diff
@@ -1,23 +1,204 @@
...
+
+For example, when a new temperature profile from a radio sonde deployment is added to a dataset of upper-air data measurements, a notification message would be published that includes the URL used to access the new temperature profile data. Everyone subscribing to notification messages about the upper-air measurement dataset would receive the notification message, identify the URL and download the new temperature profile data.
+
+Optionally, data may be embedded in a notification message using a ``content`` object _in addition_ to publishing via the data server. Embedded data must be encoded as ``UTF-8``, ``Base64``, or ``gzip``, and must not exceed 4096 bytes in length once encoded.
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:105]: base64

Embedded -> inline (aligns to WNM language)

File: guide/sections/part1/data-publisher.adoc [hunk 7]
```diff
@@ -1,23 +1,204 @@
...
+
+Finally, please note that Global Caches are under no obligation to cache data published on _experimental_ topics. For such data, the ``cache`` property should be set to ``false``.
+
+===== Implementing access control
+
+Recommended data, as defined in the WMO Unified Data Policy (Resolution 1 (Cg-Ext 2021)), is exchanged on WIS2 in support of Earth system monitoring and prediction efforts and _may_ by provided with conditions on use. This means that you may control access to Recommended data.
+
+Access control should use only the "security schemes" for authentication and authorization specified in OpenAPIfootnote:[OpenAPI security schemes https://swagger.io/docs/specification/authentication/].
```
Comment (tomkralidis) [inline guide/sections/part1/data-publisher.adoc:154]: Suggest using https://spec.openapis.org/oas/v3.1.0#security-scheme-object as reference

File: guide/sections/part2/wis2node.adoc [hunk 8]
```diff
@@ -4,136 +4,71 @@
...
 
-====== Publication and topic selection
+*Token 2* is a descriptive name for the centre and this may contain dashes (but not other special characters).
+ 
+The descriptive name should be something recognisable - not only by our community, but by other users too. Basing things on the Web domain name is likely to ensure that centre identifiers remain unique within a particular country/territory. A UK example this time: the UK's NMS is the Met Office (http://www.metoffice.gov.uk), so “metoffice” is better than “ukmo”footnote:[The “.gov” part of the domain name is superfluous for the purposes of WIS. There is nothing preventing its use, but it doesn’t add any value.]. Using the 4-letter GTS centre identifiers (*CCCC*) is not recommended because people unfamiliar with the GTS do not understand them.
```
Comment (tomkralidis) [inline guide/sections/part2/wis2node.adoc:47]: WIS or WIS2?

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Review by efucile — APPROVED** [review]

File: guide/sections/part1/data-consumer.adoc [hunk 9]
```diff
@@ -1,31 +1,93 @@
...
+A Data Consumer can find the appropriate topic to subscribe to either by searching the Global Discovery Catalogue, or by browsing the topic hierarchy on a message broker.
+
+WIS2 uses Global Caches to distribute Core data, as defined in the WMO Unified Data Policy (Resolution 1 (Cg-Ext 2021)). Each Global Cache re-publishes Core data on its own highly-available data server and publishes a new notification message advertising the availability of that data from a the Global Cache location. 
+
+Notifications from WIS2 Nodes and Global Caches are published on different topics: The root topic used by WIS2 Nodes is ``origin``, while the root topic used by Global Caches is ``cache``. Other than the root, the topic hierarchy is identical. For example, for synoptic weather observations published by Environment Canada:
+
+* Environment Canada's WIS2 Node publishes to: ``origin/a/wis2/ca-eccc-msc/data/core/weather/surface-based-observations/synop``
```
[the remark this answers is not in the document]
Comment (6a6d74) [reply guide/sections/part1/data-consumer.adoc]: Fixed.

File: guide/sections/part1/data-publisher.adoc (hunk 1 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:34]: Fixed.

File: guide/sections/part1/data-publisher.adoc (hunk 2 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:36]: Fixed.

File: guide/sections/part1/data-publisher.adoc (hunk 3 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:64]: Fixed.

File: guide/sections/part1/data-publisher.adoc (hunk 4 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:70]: Fixed.

File: guide/sections/part1/data-publisher.adoc (hunk 5 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:90]: Fixed.

File: guide/sections/part1/data-publisher.adoc (hunk 7 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:154]: Fixed. Will also clean up references to this in part2/operations.adoc and part2/wis2node.adoc

File: guide/sections/part1/data-publisher.adoc (hunk 6 above)
Comment (6a6d74) [reply guide/sections/part1/data-publisher.adoc:105]: Fixed.

**Pull request merged**